coderd/workspaceapps.go

```diff
@@ -39,6 +39,23 @@ const (
 	redirectURIQueryParam     = "redirect_uri"
 )
 
+// nonCanonicalHeaders is a map from "canonical" headers to the actual header we
+// should send to the app in the workspace. Some headers (such as the websocket
+// upgrade headers from RFC 6455) are not canonical according to the HTTP/1
+// spec. Golang has said that they will not add custom cases for these headers,
+// so we need to do it ourselves.
+//
+// Some apps our customers use are sensitive to the case of these headers.
+//
+// https://github.com/golang/go/issues/18495
+var nonCanonicalHeaders = map[string]string{
```

Comment:
nit: Only these keys or every `Sec-Websocket-*`? If the latter one, maybe you can refactor it to the future-proof logic (`Sec-Websocket-Foo`, `Sec-Websocket-Bar`, etc.)

Reply:
I considered using a regex or some custom function to do it but decided against it as the websocket spec was made in 2011 and they haven't added any new headers that require this level of massaging since. Hashmap is the fastest so I think it's a fine trade off

Reply:
Fair enough, feel free to ship it